### PR TITLE
Send will when client connection stops

### PR DESCRIPTION
**Pull request by karenbajador**
This is a fix to issues raised in the following threads:
https://github.com/rabbitmq/rabbitmq-web-mqtt/issues/23
https://groups.google.com/forum/#!searchin/rabbitmq-users/last$20will%7Csort:date/rabbitmq-users/Jg88NdgcNPY/1w7LzK10CQAJ

The change is committed so that behaviour will be similar to the rabbitmq-mqtt plugin wherein the lastwill message is dispatched by broker when the mqtt client is killed or died ungracefully.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@karenbajador Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-web-mqtt&pullRequestId=52)!

[Click here](https://cla.pivotal.io/sync/rabbitmq?repositoryId=rabbitmq/rabbitmq-web-mqtt&pullRequestId=52) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@karenbajador Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/rabbitmq?repositoryId=rabbitmq/rabbitmq-web-mqtt&pullRequestId=52)!

**Comment by michaelklishin** [comment]
Thank you. This looks good at first glance but we'd need to add an integration test before merging (our team will take care of that).

**Comment by karenbajador** [comment]
> Thank you. This looks good at first glance but we'd need to add an integration test before merging (our team will take care of that).

Thank you Michael. Looking forward to the fix. Cheers!

**Comment by michaelklishin** [comment]
I can confirm that the LWT message is sent with this patch but integration tests turned out to be not entirely trivial. On the positive side this PR lead to some testing infrastructure improvements in this plugin 👍

**Review by michaelklishin — APPROVED** [review]

**Comment by michaelklishin** [comment]
Backported to `v3.7.x`.

**Comment by michaelklishin** [comment]
Available in [3.7.10-alpha.26](https://bintray.com/rabbitmq/all-dev/rabbitmq-server/3.7.10-alpha.26). @karenbajador would you be interested in giving that build a try? We will produce a new RC very soon but it should only have one more unrelated change.

**Comment by karenbajador** [comment]
Thanks @michaelklishin . That is awesome work. We will give the build a go.

**Comment by michaelklishin** [comment]
Doh, the link is already dead because a more recent 3.7.10 alpha build was published. Please grab the most recent 3.7.10-alpha from [this Bintray repo](https://bintray.com/rabbitmq/all-dev/rabbitmq-server/).

**Comment by karenbajador** [comment]
This build looks good. Looking forward to the release. Thanks.

**Comment by michaelklishin** [comment]
[3.7.10](https://github.com/rabbitmq/rabbitmq-server/releases/tag/v3.7.10) is out.

**Comment by karenbajador** [comment]
Thanks @michaelklishin. Will the docker image be released soon as well?

**Comment by michaelklishin** [comment]
The Docker image is not maintained by our team. As far as I know, they are [blocked on how they provision Erlang](https://github.com/docker-library/rabbitmq/issues/295).

**Comment by karenbajador** [comment]
Thanks for the update!

**Pull request merged**
